src/lang/KclSingleton.ts

```diff
@@ -640,25 +646,65 @@ const defaultSelectionFilter: EntityType_type[] = [
 
 /** TODO: This function is not synchronous but is currently treated as such */
 function setSelectionFilterToDefault(
-  programMemory: ProgramMemory,
-  engineCommandManager: EngineCommandManager
+  engineCommandManager: EngineCommandManager,
+  restorePreviousSelections?: Selections
 ) {
   // eslint-disable-next-line @typescript-eslint/no-floating-promises
-  setSelectionFilter(defaultSelectionFilter, engineCommandManager)
+  setSelectionFilter(
+    defaultSelectionFilter,
+    engineCommandManager,
+    restorePreviousSelections
+  )
 }
 
 /** TODO: This function is not synchronous but is currently treated as such */
 function setSelectionFilter(
   filter: EntityType_type[],
-  engineCommandManager: EngineCommandManager
+  engineCommandManager: EngineCommandManager,
+  restorePreviousSelections?: Selections
 ) {
-  // eslint-disable-next-line @typescript-eslint/no-floating-promises
-  engineCommandManager.sendSceneCommand({
-    type: 'modeling_cmd_req',
-    cmd_id: uuidv4(),
-    cmd: {
-      type: 'set_selection_filter',
-      filter,
-    },
+  const { engineEvents } = restorePreviousSelections
+    ? handleSelectionBatch({
+        selections: restorePreviousSelections,
+      })
+    : { engineEvents: undefined }
+  if (!restorePreviousSelections || !engineEvents) {
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    engineCommandManager.sendSceneCommand({
+      type: 'modeling_cmd_req',
+      cmd_id: uuidv4(),
+      cmd: {
+        type: 'set_selection_filter',
```

Comment:
[q] does `set_selection_filter` clear the selection? That's interesting and seems not correct

Reply:
Yes that's right, did seem a little odd, maybe because the old selections could be illegal once the new selection filters were applied, but always nuking them does seem odd, but last thing I wanted to do was go back and forth with engine team when getting this ready for demo.